apis/python/setup.py

```diff
@@ -117,7 +117,6 @@ def run(self):
     extras_require={
         "dev": [
             "black",
-            "flake8-bugbear",
```

Comment:
I had/have in mind of making pre-commit the single entry point to all linters and remove most entries from here. This can go in a separate PR, I added ruff [here](https://github.com/single-cell-data/TileDB-SOMA/pull/751/commits/5c7a4fd90b95de624d60434bd27b3758639a350f) for now.